plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -45,51 +45,41 @@ const getImageIndex = ( context: ProductGalleryContext, imageId: string ) => {
 	return imagesIds.indexOf( imageId );
 };
 
-const getImageId = ( context: ProductGalleryContext, imageIndex: number ) => {
-	const imagesIds = getCurrentImages( context );
-
-	if ( imageIndex < 0 ) {
-		return imagesIds.at( 0 ) || '';
-	}
-
-	if ( imageIndex > imagesIds.length - 1 ) {
-		return imagesIds.at( -1 ) || '';
-	}
-
-	return imagesIds[ imageIndex ];
-};
-
 const disableArrows = (
 	context: ProductGalleryContext,
 	nextImageIndex: number
 ) => {
 	const imagesIds = getCurrentImages( context );
-	context.disableLeft = nextImageIndex === 0;
-	context.disableRight = nextImageIndex === imagesIds.length - 1;
+	context.disableLeft = nextImageIndex === 1;
+	context.disableRight = nextImageIndex === imagesIds.length;
 };
 
 const selectImage = (
 	context: ProductGalleryContext,
-	type: 'prev' | 'next' | 'current'
+	type: 'prev' | 'next' | 'current' | 'first'
 ) => {
-	const selectedImageIdIndex = getImageIndex(
-		context,
-		context.selectedImage
-	);
+	const { selectedImageIndex, imageId, dialogVisibleImagesIds } = context;
+	let newImageIndex = 1;
 
-	// explicit "current"
-	let nextPotentialIndex = selectedImageIdIndex;
+	if ( type === 'first' ) {
+		newImageIndex = 1;
+	}
+
+	if ( type === 'current' ) {
+		newImageIndex = getImageIndex( context, imageId ) + 1;
```

Comment:
It looks like we use both zero-based (e.g. in `disableArrows`) and one-based (here) indexing. Caught me out when reading the code. Ideally we'd be consistent.

Reply:
Yeah, so the reason for using one-based is I'm using `selectedImageIndex` as the actual value that gets rendered on the frontend via `data-wc-text` directive [here](https://github.com/woocommerce/woocommerce/pull/54570/files#diff-646bdfbca017740548f7a8a7218bd10914aaac440748cc747d9052e3d803b7a9R69):

```
<span class="wc-block-product-gallery-pager__current-index" data-wc-text="context.selectedImageIndex"></span>
```

Perhaps changing the variable's name to `selectedImageNumber` and corresponding variables (e.g. `newImageIndex` -> `newImageNumber`) would be good move? Plus comments.

Unfortunately, `data-wc-text` directive doesn't allow modifying the output which would be ideal.

Reply:
Thanks for the explanation! I wouldn't consider this a blocker but just something I noticed that could potentially improved at the time